src/metabase/util/malli.cljc

```diff
@@ -92,27 +93,36 @@
                                                             "\n"
                                                             (str "\n          "))
                                 (when (not-empty doc) (str "\n\n  " doc))))
-           id (str (gensym "id"))]
-       `(let [defn# (core/defn
-                      ~name
-                      ~@(some-> annotated-doc vector)
-                      ~(assoc meta
-                              :raw-arglists (list 'quote raw-arglists)
-                              :schema schema
-                              :validate! id)
-                      ~@(map (fn [{:keys [arglist prepost body]}] `(~arglist ~prepost ~@body)) parglists)
-                      ~@(when-not single (some->> arities val :meta vector)))]
-          (mc/=> ~name ~schema)
-          ;; instrument the defn we just registered, via ~id
-          (instrument! ~id)
-          defn#))))
+           id (str (gensym "id"))
+           inner-defn `(core/defn
+                         ~name
+                         ~@(some-> annotated-doc vector)
+                         ~(assoc meta
+                                 :raw-arglists (list 'quote raw-arglists)
+                                 :schema schema
+                                 :validate! id)
+                         ~@(map (fn [{:keys [arglist prepost body]}] `(~arglist ~prepost ~@body)) parglists)
+                         ~@(when-not single (some->> arities val :meta vector)))]
+       (case target
+         :clj  `(let [defn# ~inner-defn]
+                  (mc/=> ~name ~schema)
+                  ;; instrument the defn we just registered, via ~id
+                  (instrument! ~id)
+                  defn#)
+         ;; Vars aren't real in CLJS, so wrapping the inner `defn` in a `let` doesn't work like in does in CLJ.
+         ;; In CLJS `mu/defn` is just `cljs.core/defn` with some extra metadata and augmented docstring;
+         ;; [[mc/=>]] is not called, nor is [[instrument!]].
+         :cljs inner-defn))))
 
 #?(:clj
    (defmacro defn
      "Like s/defn, but for malli. Will always validate input and output without the need for calls to instrumentation (they are emitted automatically).
      Calls to minst/unstrument! can remove this, so use a filter that avoids :validate! if you use that."
      [& args]
-     (-defn mx/SchematizedParams args)))
+     ;; [[macros/case]] only works properly in a `defmacro`, not in a helper function called by a `defmacro`.
+     ;; So we use it here and pass :clj or :cljs to [[-defn]].
+     (-defn (macros/case :clj :clj :cljs :cljs)
```

Comment:
since this is wrapped with the `#?(:clj ...` reader macro, will this always going to be :clj in here?